### PR TITLE
Vision

### PR DESCRIPTION
**Pull request by GilStein1**
image

### CODE REVIEW

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Alliance.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package org.firstinspires.ftc.teamcode;
+
+public enum Alliance {
+
+    BLUE,
+    RED;
```
Comment (YoniKiriaty) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Alliance.java]: pick convention, ; or nothing and same line or not

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/robotstatetesters/RazRobotStateTest.java [hunk 2]
```diff
@@ -15,6 +16,11 @@ public void initialize() {
 
     }
 
+    @Override
+    public Alliance setAlliance() {
+        return Alliance.RED;
+    }
```
Comment (YoniKiriaty) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/robotstatetesters/RazRobotStateTest.java]: remove

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/subsystemstesters/RazVisionTest.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+    @Override
+    public void execute() {
+        Robot.getInstance().getVision().telemetry(telemetry);
+    }
+
+    @Override
+    public void configureBindings() {
+
+    }
```
Comment (YoniKiriaty) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/subsystemstesters/RazVisionTest.java:25]: spacve

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/vision/Vision.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+public class Vision extends SubsystemBase {
+
+    private final AprilTagProcessor aprilTagProcessor;
+    private final TfodProcessor tfodProcessor;
+    private final PropProcessor propProcessor;
+    private final VisionPortal visionPortal;
+
+    public Vision(HardwareMap hardwareMap) {
+
```
Comment (YoniKiriaty) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/vision/Vision.java]: remove space

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/vision/Vision.java [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+                .setCameraResolution(new Size(VisionConstant.CAMERA_RESOLUTION_WIDTH, VisionConstant.CAMERA_RESOLUTION_HEIGHT))
+                .setStreamFormat(VisionConstant.STREAM_FORMAT)
+                .enableLiveView(true)
+                .setAutoStopLiveView(true)
+                .build();
+    }
+
+    public List<AprilTagDetection> getTagsDetections() {
+        return aprilTagProcessor.getDetections();
+    }
```
Comment (YoniKiriaty) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/vision/Vision.java:54]: space

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/DefaultRaz.java [hunk 6]
```diff
@@ -3,12 +3,14 @@
...
 import com.qualcomm.robotcore.eventloop.opmode.OpMode;
 
+import org.firstinspires.ftc.teamcode.Alliance;
 import org.firstinspires.ftc.teamcode.Robot;
 
 public abstract class DefaultRaz extends OpMode {
 
     @Override
     public final void init() {
+        Robot.getInstance().setAlliance(setAlliance());
```
Comment (YoniKiriaty) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/DefaultRaz.java]: rwmpve

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/DefaultRaz.java [hunk 7]
```diff
@@ -20,6 +22,10 @@ public final void loop() {
         execute();
     }
 
+    public Alliance setAlliance() {
+        return Alliance.RED;
+    }
```
Comment (YoniKiriaty) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/opmodes/DefaultRaz.java]: rempve

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Location.java [hunk 8]
```diff
@@ -0,0 +1,7 @@
+package org.firstinspires.ftc.teamcode;
+
+public enum Location {
+
```
Comment (bossyshrimp68) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Location.java:4]: keep the convention

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/vision/PropProcessor.java [hunk 9]
```diff
@@ -0,0 +1,93 @@
...
+    private Location location;
+
+    @Override
+    public void init(int width, int height, CameraCalibration calibration) {
+
+    }
+
+    @Override
+    public Object processFrame(Mat frame, long captureTimeNanos) {
+
```
Comment (greenblitz4590) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/vision/PropProcessor.java]: space

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot.java [hunk 10]
```diff
@@ -18,33 +18,41 @@ public static void init(HardwareMap hardwareMap) {
         getInstance().initSubsystems(hardwareMap);
     }
 
+    private Robot() {
+        this.alliance = Alliance.RED;
+    }
```
Comment (hillelv) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/Robot.java]: move under alliamnce

File: TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/vision/Vision.java [hunk 11]
```diff
@@ -0,0 +1,81 @@
...
+
+    public List<Pair<AprilTagPoseFtc,Integer>> getRelativeTagsPoses() {
+        List<AprilTagDetection> temp = getTagsDetections();
+        List<Pair<AprilTagPoseFtc,Integer>> poses = new ArrayList<>();
+        for(AprilTagDetection detection : temp)
+            if(detection.metadata != null)
+                poses.add(new Pair<>(detection.ftcPose,detection.id));
```
Comment (hillelv) [inline TeamCode/src/main/java/org/firstinspires/ftc/teamcode/subsystems/vision/Vision.java]: {{}}

**Review by hillelv — APPROVED** [review]

**Pull request merged**